wasmi_v1/src/func/typed_func.rs

```diff
@@ -134,7 +134,7 @@ where
 ///
 /// [`Engine`]: [`crate::Engine`].
 pub struct CallResultsTuple<Results> {
-    _marker: PhantomData<fn() -> Results>,
+    _marker: PhantomData<Results>,
```

Comment:
This change is incorrect since `CallResultsTuple` does not have owning semantics over `Results`. Please revert.